vm_insnhelper.c

```diff
@@ -914,15 +914,8 @@ static inline VALUE
 vm_get_cbase(const VALUE *ep)
 {
     const rb_cref_t *cref = vm_get_cref(ep);
-    VALUE klass = Qundef;
-
-    while (cref) {
-	if ((klass = CREF_CLASS(cref)) != 0) {
-	    break;
-	}
-	cref = CREF_NEXT(cref);
-    }
```

Comment:
@ko1 Can `CREF_CLASS(cref)` be no longer zero? According to the code coverage, this loop seems indeed unused.

https://rubyci.s3.amazonaws.com/coverage-latest-html/ruby/vm_insnhelper.c.gcov.html#920

![image](https://user-images.githubusercontent.com/21557/142745047-208a19d9-eca9-4bcc-bf16-dfbaf5c3409a.png)

Reply:
Thanks for bringing this up. I couldn't find anywhere this happened (even before this change) and added an assertion to `cref_new` https://github.com/ruby/ruby/pull/5146/files#diff-2af2e7f2e1c28da5e9d99ad117cba1c4dabd8b0bc3081da88e414c55c6aa9549R251

Reply:
I have no idea... Let's try!